src/base/static/state/ducks/datasets.js

```diff
@@ -0,0 +1,85 @@
+// Selectors:
+export const datasetSelector = (state, datasetSlug) =>
+  state.datasets.find(dataset => dataset.slug === datasetSlug);
+
+// Actions:
+const LOAD = "datasets/LOAD";
+
+// Action creators:
+export function loadDatasets(datasets) {
+  const getTagChildren = (dataset, node) => {
+    const childNodes = [];
+
+    const traverse = (dataset, node) => {
+      if (node.parent) {
+        const newNode = dataset.tags.find(
+          childTag => childTag.id === node.parent,
+        );
+        childNodes.push(newNode);
+
+        traverse(dataset, newNode);
+      }
+    };
+    traverse(dataset, node);
+
+    // Traversing the tag tree produces an array in backward order, so
+    // return the reversed array.
+    return childNodes.reverse();
+  };
+
+  datasets = datasets.map(dataset => {
+    dataset.tags = dataset.tags
+      .map(({ is_enabled: isEnabled, ...rest }) => ({
+        isEnabled,
+        ...rest,
+      }))
+      .map(tag => {
+        const children = getTagChildren(dataset, tag);
+
+        tag.children = children;
+        tag.displayName =
+          children.length > 0
+            ? children.map(child => child.name).concat([tag.name])
```

Comment:
This ternary is not needed. If `children` is an empty array, then `children.map(child => child.name).concat([tag.name])` is equal to `[tag.name]`.